docs/how_to_guides/miscellaneous/how_to_add_and_test_a_new_sqlalchemydataset_class.rst

```diff
@@ -0,0 +1,215 @@
+                from pyhive.sqlalchemy_presto import PrestoDialect as prestodialect
+
+                PRESTO_TYPES = {
+                    "VARCHAR": sqltypes.VARCHAR,
+                    "TEXT": sqltypes.VARCHAR,
+                    "CHAR": sqltypes.CHAR,
+                    "DOUBLE": prestotypes.DOUBLE,
+                    "INTEGER": sqltypes.INTEGER,
+                    "SMALLINT": sqltypes.SMALLINT,
+                    "BIGINT": sqltypes.BIGINT,
+                    "DATETIME": sqltypes.TIMESTAMP,
+                    "TIMESTAMP": sqltypes.TIMESTAMP,
+                    "DATE": sqltypes.DATE,
+                    "FLOAT": prestotypes.DOUBLE,
+                    "BOOLEAN": prestotypes.BOOLEAN,
+                }
+            except ImportError:
+                PRESTO_TYPES = {}
+
+    * **In get_dataset, add an elif clause to connect to data, .** Note: some SQL backends require a schema. If so, this is also the right place to create a test schema, usually called ``test_ci``.
+
+        .. code-block:: python
+
+            elif dataset_type == "presto":
+                from sqlalchemy import create_engine
+
+                engine = create_engine("presto://presto@localhost/memory/test_ci", echo=False)
+                conn = engine.connect()
+
+                sql_dtypes = {}
+                if (
+                    schemas
+                    and "presto" in schemas
+                    and isinstance(engine.dialect, prestodialect)
+                ):
+                    schema = schemas["presto"]
+                    sql_dtypes = {col: PRESTO_TYPES[dtype] for (col, dtype) in schema.items()}
+                    for col in schema:
+                        type_ = schema[col].lower()
+                        if type_ in ["integer", "smallint", "bigint"]:
+                            df[col] = pd.to_numeric(df[col])
+                        elif type_ in ["float", "double"]:
+                            df[col] = pd.to_numeric(df[col])
+                        elif type_ in ["timestamp", "datetime"]:
+                            df[col] = pd.to_datetime(df[col]).dt.strftime('%Y-%m-%d %H:%M:%S')
+                        elif type_ in ["varchar"]:
+                            df[col] = df[col].astype(str)
+
+                tablename = generate_test_table_name(dataset_id)
+
+                conn.execute("CREATE SCHEMA IF NOT EXISTS test_ci")
+                df.to_sql(name=tablename, con=conn, index=False)
+
+                # Build a SqlAlchemyDataset using that database
+                return SqlAlchemyDataset(
+                    tablename, engine=conn, profiler=profiler, caching=caching
+                )
+
+
+    * **Add your dialect to candidate_test_is_on_temporary_notimplemented_list**.
+
+        .. code-block:: python
+
+            def candidate_test_is_on_temporary_notimplemented_list(context, expectation_type):
+                if context in ["sqlite", "postgresql", "mysql", "presto"]:
+
+5. Get tests to pass
```

Comment:
New section: 
Make changes to test definitions 
-> describe adding the dialect to only_for or "suppress_test_for"

Reply:
IMO, we should discourage contributors from using `only_for` and `suppress_test_for`, since they create exceptions to the "works the same on all execution engines" promise we're trying to adhere to.

Proposal: under "Get tests to pass", add this blurb:


In some rare cases, you may need to suppress certain tests for your SQL backend. In that case, you can use the `only_for` or `suppress_test_for` flags in the test configs. However, we try very hard to avoid such cases, since they weaken the "works the same on all execution engines" principle of Great Expectations.

Reply:
I like that language for only_for and suppress_test_for! The contributor should also be adding any tests that are needed to highlight dialect-specific quirks.

Instead of "Get tests to pass" how about "Use tests to verify [semantic] consistency".  I *really* hate it when people think of tests as a guarantee that a thing is right, and "get tests to pass" makes it sound like if the tests pass then we think it's right. Tests passing is necessary but not sufficient.